agent/pkg/crane/k8s/configmap.go

```diff
@@ -25,8 +25,8 @@ func newConfigmap(ctx context.Context) *configmap {
 	return &configmap{ctx: ctx, status: "", avail: []string{}}
 }
 
-func (cm *configmap) loadSharedConfig(namespace string) error {
-	client, err := getConfigMapClient(namespace)
+func (cm *configmap) loadSharedConfig(namespace string, cfg *config.Configuration) error {
```

Comment:
Config should be on the configmap facade/wrapper,  only needing us to pass on instantiation.